### ISSUE TITLE
Discourage Roads with tracks on them

### ISSUE THREAD

**Issue by abhumbla** [bikehopper/graphhopper#76]
try not to route down them (if there's no protected bike infra) and be careful with turns. not quite sure how to formalize this, just don't like this routing down church
![Screenshot_20220429-162835](https://user-images.githubusercontent.com/9853610/166080492-f4158f4f-aeef-4202-abd1-34f1d57219f3.png)

**Comment by hcourt** [comment]
The way this is mapped on this stretch is with `railway=level_crossing` nodes.

![image](https://user-images.githubusercontent.com/2773087/167222067-15fce86e-3b72-4904-b38d-f6cd2599ff63.png)

**Comment by abhumbla** [comment]
Thinking about this, we don't need to discourage roads with tracks if there are other lanes like @graue was saying, but turning onto a road with tracks is potentially hazardous so maybe we could incorporate this into the turn cost stuff you were working on @hcourt ?

**Comment by hcourt** [comment]
@abhumbla yes, I think so. It's at least related and depending on how these crossings are processed into edges, might be easy to add into that code.

That and this issue kinda brings up something I've been thinking for turn costs, that in addition to ways having encoded penalties that affect the weight (BikeCommonFlagEncoder et al), nodes need that too. 

Currently there is one (1) way that node tags affect routing, which is if the node is a barrier then don't route through it. And it's done pretty hackily, by creating artificial ways split at the barrier, adding artificial tags to that way, and then saying access=no or whatever on the way. We could replicate/piggyback on that hack for track crossings & signals (for turn costs).  I'm a little eager (perhaps too much) to sink into the mud of adding per-node penalties though lol. Talk me down from writing that refactor?

**Comment by abhumbla** [comment]
I trust you, go for it if you think it'll make the code less hacky and conceptually easier to understand
